cEP-0027.md

```diff
@@ -0,0 +1,144 @@
+# coala Bears Testing API
+
+| Metadata |                                           |
+| -------- | ----------------------------------------- |
+| cEP      | 27                                        |
+| Version  | 0.1                                       |
+| Title    | coala Bears Testing API                   |
+| Authors  | Sangam Kumar <mailto:ksangam68@gmail.com> |
+| Status   | Proposed                                  |
+| Type     | Feature                                   |
+
+## Abstract
+
+This cEP describes the process of adding `BaseTestHelper` class and
+`GlobalBearTestHelper` class to improve the testing API of coala bears, as a
+part of the [GSoC 2018 project](https://summerofcode.withgoogle.com/projects/#6625036551585792).
+
+## Introduction
+
+[coala](https://coala.io) has its own testing API which includes
+`LocalBearTestHelper`, a helper class for testing of local bears. It does not
+provide support for `GlobalBear`. Writing tests for a `GlobalBear` is slightly
+lengthy for a developer. Example:
+[GitCommitBearTest](https://github.com/coala/coala-bears/blob/master/tests/vcs/git/GitCommitBearTest.py),
+[VultureBearTest](https://github.com/coala/coala-bears/blob/master/tests/python/VultureBearTest.py),
+etc. `GlobalBearTestHelper` is going to be a helper class for simplification of
+testing of `GlobalBear`.
+
+Also, the testing API should have a base test helper class which should be
+inherited in all tests' helper (LocalBearTestHelper, GlobalBearTestHelper).
+This project introduces `BaseTestHelper` class, base class for all bear tests.
+
+## Proposed Approach
+
+### 1. Implementation of the `BaseTestHelper` class
+
+Issue: [coala/coala#3676](https://github.com/coala/coala/issues/3676)
+
+As mentioned in [introduction](#Introduction), `BaseTestHelper` class is the
+base class for all bear tests.
+
+```py
+class BaseTestHelper(object):
```

Comment:
We are Python3, no need to explicitly inherit `object` ;) Anyway, can be changed after merge